router/xgress/xgress.go

```diff
@@ -188,16 +194,26 @@ func (self *Xgress) firstSessionStartReceived() bool {
 
 func (self *Xgress) Start() {
 	log := pfxlog.ContextLogger(self.Label())
-	if !self.IsTerminator() {
+	if self.IsTerminator() {
+		log.Debug("terminator: waiting for session start before starting receiver")
+		if self.Options.SessionStartTimeout > time.Second {
+			time.AfterFunc(self.Options.SessionStartTimeout, self.terminateIfNotStarted)
+		}
+	} else {
 		log.Debug("initiator: sending session start")
 		self.forwardPayload(self.GetStartSession())
 		go self.rx()
-	} else {
-		log.Debug("terminator: waiting for session start before starting receiver")
 	}
 	go self.tx()
 }
 
+func (self *Xgress) terminateIfNotStarted() {
+	if self.IsSessionStarted() {
```

Comment:
Shouldn't this be `!self.IsSessionStarted()`?

Reply:
yeah, noticed this in testing. fixed